src/problem3.rb

```diff
@@ -0,0 +1,17 @@
+class Problem3
+  def getDuplicatedValues(values)
+    response = []
+    hash = Hash.new(-1)
```

Comment:
Still not there,

```
Do not sort the array or use bit vectors. Try to work within the array without using any other temporary data structures
```

Hash can qualify both as a bit vector and as a temporary data structure.